frontend/packages/dev-console/src/components/topology/shapes/PodSet.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { get } from 'lodash';
+import {
+  PodStatus,
+  calculateRadius,
+  getDataBasedOnCurrentAndPreviousRC,
+  rollback,
+} from '@console/shared';
+import { k8sUpdate } from '@console/internal/module/k8s';
+import { DeploymentConfigModel } from '@console/internal/models';
+import { DonutStatusData } from '../topology-types';
+
+interface PodSetProps {
+  size: number;
+  data: DonutStatusData;
+}
+
+const calculateInnerPodStatusRadius = (
+  outerPodStatusInnerRadius: number,
+  outerPodStatusWidth: number,
+): { innerPodStatusOuterRadius: number; innerPodStatusInnerRadius: number } => {
+  const innerPodStatusWidth = outerPodStatusWidth * 0.6;
+  const spaceBwOuterAndInnerPodStatus = 3;
+  const innerPodStatusOuterRadius = outerPodStatusInnerRadius - spaceBwOuterAndInnerPodStatus;
+  const innerPodStatusInnerRadius = innerPodStatusOuterRadius - innerPodStatusWidth;
+
+  return { innerPodStatusOuterRadius, innerPodStatusInnerRadius };
+};
+
+const PodSet: React.FC<PodSetProps> = ({ size, data }) => {
+  const { podStatusOuterRadius, podStatusInnerRadius, podStatusStrokeWidth } = calculateRadius(
+    size,
+  );
+  const { innerPodStatusOuterRadius, innerPodStatusInnerRadius } = calculateInnerPodStatusRadius(
+    podStatusInnerRadius,
+    podStatusStrokeWidth,
+  );
+  const currentRCPhase = get(data.current, 'metadata.name');
+  if (currentRCPhase && currentRCPhase === 'Failed') {
+    rollback(data.dc, data.previous)
+      .then((res) => {
+        k8sUpdate(DeploymentConfigModel, res, res.metadata.namespace, res.metadata.name);
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.log('Error in rollback', error);
```

Comment:
Can we capture this error better than console.log?

Reply:
@rohitkrai03 Currently, this side effect is in a discussion. We might not even need this.

Reply:
Firstly, you're calling `rollback` inside the render function which is bad.
Secondly why is this needed? Seems odd for this piece of UI code to be making calls that change resources without the user invoking any action.
If we share the same logic as the workload overview, doesn't the pod rings show the correct data?

Reply:
@christianvogt I'm working on removing it and use the same logic from the workload overview.